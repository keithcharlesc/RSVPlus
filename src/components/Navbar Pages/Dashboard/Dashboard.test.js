import React from "react";
import { render, getByTestId } from "@testing-library/react";
import { toHaveTextContent } from "@testing-library/jest-dom";
import TestRenderer from "react-test-renderer";
import MockDashBoard from "./MockDashBoard";

it("renders without crashing", () => {
  const { getByText } = render(<MockDashBoard />);
  getByText("DASHBOARD");
  expect(getByTestId(document.documentElement, "buttonOne")).toHaveTextContent(
    "DISPLAY UPCOMING EVENTS"
  );
});

it("matches snapshot", () => {
  const tree = TestRenderer.create(<MockDashBoard />).toJSON();
  expect(tree).toMatchSnapshot();
});
