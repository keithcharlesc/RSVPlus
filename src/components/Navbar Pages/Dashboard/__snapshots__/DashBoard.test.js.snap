// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="bg-dark"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="nav-bar-appName text-danger navbar-brand"
        href="/"
      >
        RSVP+
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/channels"
            href="/channels"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Channels
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/calendar"
            href="/calendar"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Calendar
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/faq"
            href="/faq"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            FAQ
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/contact-us"
            href="/contact-us"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Contact Us
          </a>
        </div>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="nav-bar-name navbar-text"
          >
            Signed in as: 
            Bob
            <img
              className="photo ml-2 rounded-circle"
              src="thttp://email.com/image"
            />
          </span>
        </div>
      </div>
      <button
        className="ml-3 btn btn-outline-primary"
        data-testid="button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Log Out
      </button>
    </nav>
    <div>
      
    </div>
  </div>
  <div
    className="p-3 mb-2 bg-dark text-white"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <h2
      className="page-header text-center mb-4"
    >
      DASHBOARD
    </h2>
    <div
      className="container-fluid"
    >
      <div
        className="d-flex align-items-center justify-content-center mb-4 row"
      >
         
        <button
          className="d-flex align-items-center justify-content-center button"
          data-testid="buttonOne"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "height": 30,
              "width": 250,
            }
          }
        >
          <span
            className="button__text"
          >
            DISPLAY UPCOMING EVENTS
          </span>
        </button>
      </div>
      <div
        className="row"
      >
        <div
          className="d-flex justify-content-center container"
        >
          <div
            className="events"
          />
        </div>
        <div
          className="row"
        >
          <div
            className="d-flex justify-content-center container-fluid"
          >
            <div>
              <div
                className="card"
              >
                <div
                  className="text-white h6 card-body"
                >
                  <small>
                    <p>
                      Hey, thanks for visiting RSVP+! ðŸ˜ƒ
                    </p>
                    <p
                      className="mt-4"
                    >
                      To get started on how to use RSVP+, check out our user guide (poster) using either one of these links !
                    </p>
                    <ul>
                      <li>
                        <a
                          className="mr-2"
                          href="https://www.figma.com/file/zI65CDPZoYCcvUK7Bw4GuZ/RSVPlus-Milestone-2?node-id=0%3A1"
                          rel="noreferrer"
                          target="_blank"
                        >
                          Figma
                        </a>
                        <a
                          href="https://drive.google.com/file/d/1iSB9UaBqE0AyVJVUmLtc2BbF1cqWHY7f/view?usp=sharing"
                          rel="noreferrer"
                          target="_blank"
                        >
                          Google Drive
                        </a>
                      </li>
                    </ul>
                    <strong>
                      Features for each tab! ðŸ”¥
                    </strong>
                    <ol>
                      <li
                        className="mt-4 mb-1"
                      >
                        Home (Dashboard)
                        <ul>
                          <li
                            className="mt-2"
                          >
                            Display upcoming events from your Google Calendar to see what's on your schedule! (Does not show past events)
                          </li>
                        </ul>
                      </li>
                      <li
                        className="mb-1 mt-4"
                      >
                        Channels
                        <ul>
                          <li
                            className="mt-2"
                          >
                            Create your own channel for a meet-up or event and invite your friends of RSVP+. [Host]
                          </li>
                          <li
                            className="mt-2"
                          >
                            View channels you have been invited to or have created. [Host & User]
                          </li>
                          <li
                            className="mt-2"
                          >
                            Delete a channel that you are the host of should you wish to remove it. [Host]
                          </li>
                          <li
                            className="mt-2"
                          >
                            Derive optimal time slots for an event (channel) for specified dates and time range after everyone has "Agreed to Sync"! âœ¨ [Host & User]
                          </li>
                          <li
                            className="mt-2"
                          >
                            Join a channel by simply entering the channel ID if it has yet to determine slots. [User]
                          </li>
                          <li
                            className="mt-2"
                          >
                            Invite more people (that you might have forgotten) to the channel if it has yet to determine slots. [Host]
                          </li>
                          <li
                            className="mt-2"
                          >
                            Remove a user from a channel if the user takes too long to respond or is unable to make it in the end, slots will be updated without his/ her data! [Host]
                          </li>
                          <li
                            className="mt-2"
                          >
                            Leave a channel after everyone has decided on a date and time, or if you are unable to make it or should you wish to turn down the invitation. [User]
                          </li>
                          <li
                            className="mt-2"
                          >
                            Refresh the channel with your latest Google Calendar data should there be any new changes made to your calendar to get newly determined slots! [Host & User]
                          </li>
                        </ul>
                      </li>
                       
                      <li
                        className="mb-1 mt-4"
                      >
                        Calendar
                         
                        <ul>
                          <li
                            className="mt-2"
                          >
                            View your logged in Gmail's primary Google Calendar by saving your URL! (Guide provided)
                          </li>
                          <li
                            className="mt-2"
                          >
                            Create a custom event that will be automatically added to your Google Calendar!
                          </li>
                        </ul>
                      </li>
                       
                      <li
                        className="mb-1 mt-4"
                      >
                        FAQ
                         
                        <ul>
                          <li
                            className="mt-2"
                          >
                            View frequently asked questions or answers to questions you might have in mind!
                          </li>
                        </ul>
                      </li>
                       
                      <li
                        className="mb-1 mt-4"
                      >
                        Contact Us
                        <ul>
                          <li
                            className="mt-2"
                          >
                            A platform to give feedback to the developers!
                          </li>
                        </ul>
                      </li>
                    </ol>
                    <p
                      className="mt-4"
                    >
                      Get started by clicking the "Display Upcoming Events" button you see above if you have a Google Cal Event (later than the current time) and explore the other tabs!
                    </p>
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
